packages/cli-repl/src/cli-repl.ts

```diff
@@ -151,10 +151,41 @@ class CliRepl {
 
     const initialServiceProvider = await this.connect(driverUri, driverOptions);
     const initialized = await this.mongoshRepl.initialize(initialServiceProvider);
+    const commandLineLoadFiles = this.listCommandLineLoadFiles();
+    if (commandLineLoadFiles.length > 0 || this.cliOptions.eval !== undefined) {
+      await this.loadCommandLineFiles(commandLineLoadFiles);
+      if (!this.cliOptions.shell) {
+        await this.exit(0);
+        return;
+      }
+    }
     await this.loadRcFiles();
     await this.mongoshRepl.startRepl(initialized);
   }
 
+  listCommandLineLoadFiles(): string[] {
+    const startIndex = this.cliOptions.nodb ? 0 : 1;
+    return (this.cliOptions._ ?? []).slice(startIndex);
+  }
+
+  async loadCommandLineFiles(files: string[]) {
```

Comment:
Nit: could we rename that to `loadCommandLineFilesAndEval`? Just from reading it seems more like a "side effect" that it also loads the eval code.

Reply:
ah, sure :) will do that along with fixing the test that’s still borked up (which is due to nyc/coverage, it seems…)